src/compat/byteswap.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2014-2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef GRIDCOIN_COMPAT_BYTESWAP_H
```

Comment:
I think we should keep the Bitcoin ifndefs so we stay closer to the original source.

Reply:
I can change them back. If you are ok with it, I'll do it in another commit, because it is too hard to undo all of the above at this point...